### PR TITLE
RAD-158: Changed image units.

### LINKED ISSUE #388: Change Image Units
_Issue [RAD-158](https://jira.stsci.edu/browse/RAD-158) was created on JIRA by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe):_

Update Image model schema units for data, variances, etc. to appropriate powers of DN/s and MJy/sr.

### PR DESCRIPTION
**Pull request by PaulHuwe**
Resolves [RAD-158](https://jira.stsci.edu/browse/RAD-158)

Closes #388 

This PR changes the units of the variables within the Image Model schema. 

NOTE: no regression test link, because these changes require update to regression test files.

**Checklist**
- [ ] Schema changes discussed at RAD Review Board meeting
- [x] Added entry in `CHANGES.rst` under the corresponding subsection
- [x] Updated relevant roman_datamodels utilities and tests
- [ ] Passed romancal regression testing on Jenkins / PLWishMaster. Link: https://plwishmaster.stsci.edu:8081/job/RT/job/romancal/XXX/

### CODE REVIEW

**Review by ddavis-stsci — APPROVED** [review]

File: src/rad/resources/schemas/wfi_image-1.0.0.yaml [hunk 1]
```diff
@@ -37,8 +38,10 @@ properties:
         exact_datatype: true
         ndim: 2
       unit:
-        tag: tag:astropy.org:astropy/units/unit-1.*
-        enum: ["electron / s"]
+        oneOf:
+          - tag: tag:stsci.edu:asdf/unit/unit-1.*
+          - tag: tag:astropy.org:astropy/units/unit-1.*
+        enum: ["DN / s", "MJy.sr**-1"]
```
Comment (nden) [inline src/rad/resources/schemas/wfi_image-1.0.0.yaml:44]: @braingram Can one of the tags be dropped here? I believe DN/s is a VO unit and supported by astropy.

Comment (braingram) [reply src/rad/resources/schemas/wfi_image-1.0.0.yaml:44]: Thanks for the ping!

The `oneOf` here is needed as `asdf-astropy` will select which tag to use based on the unit (and if it is or is not a VO unit). See the [relevant converter code here](https://github.com/astropy/asdf-astropy/blob/448542b99ff6a82b9aeb810e78fb253e2fbc155f/asdf_astropy/converters/unit/unit.py#L23-L34).

Writing out a `ImageModel` with `DN / s` data produces the following and uses the astropy tag:
```yaml
  data: !unit/quantity-1.1.0
    unit: !<tag:astropy.org:astropy/units/unit-1.0.0> DN / s
    value: !core/ndarray-1.0.0
      source: 9
      datatype: float32
      byteorder: little
      shape: [4088, 4088]
```
whereas using `MJy.sr**-1` produces the following and uses the stsci tag:
```yaml
  data: !unit/quantity-1.1.0
    unit: !unit/unit-1.0.0 MJy.sr**-1
    value: !core/ndarray-1.0.0
      source: 9
      datatype: float32
      byteorder: little
      shape: [4088, 4088]
```

Comment (nden) [reply src/rad/resources/schemas/wfi_image-1.0.0.yaml:44]: Huh! My bad! I checked DN/s but not MJy/sr as I thought that surely this is a VO unit. Apparently not!

File: src/rad/resources/schemas/wfi_image-1.0.0.yaml [hunk 2]
```diff
@@ -26,9 +26,10 @@ properties:
             tag: asdf://stsci.edu/datamodels/roman/tags/source_detection-1.0.0
         required: [photometry, wcs]
   data:
-    title: Science Data (electrons / s)
+    title: Science Data (DN / s) or (MJy / sr)
```
[the remark this answers is not in the document]
Comment (PaulHuwe) [reply src/rad/resources/schemas/wfi_image-1.0.0.yaml:29]: L2, and the MJy/sr option was specifically requested in the ticket.

File: CHANGES.rst [hunk 3]
```diff
@@ -16,6 +16,8 @@
 
 - Updated product table names. [#382]
 
+- Changed image units. [#389]
```
[the remark this answers is not in the document]
Comment (PaulHuwe) [reply CHANGES.rst]: Updated.

**Review by schlafly — APPROVED** [review]
LGTM, thanks!

**Pull request merged**